### ISSUE TITLE
pkg-config --version is not a good check for whether pkg-config is usable

### ISSUE THREAD

**Issue by megous** [mesonbuild/meson#5402]
https://github.com/mesonbuild/meson/issues/4057

I use the following wrapper when cross-compiling:

```
export PKG_CONFIG_LIBDIR="/workspace/megous.com/apps-c/static-deps/output/arm64-lto/sys/usr/lib/pkgconfig"

pkg-config --define-variable=prefix=/workspace/megous.com/apps-c/static-deps/output/arm64-lto/sys/usr --static "$@" | tee -a "/workspace/megous.com/apps-c/static-deps/pkglog"
ret=${PIPESTATUS[0]}

exit $ret
```
Meson since version 0.50 or so, checks whether pkg-config can be used by calling pkg-config --version.

This fails with:

```
Pkg-config binary for MachineChoice.HOST is not cached.
Pkg-config binary for MachineChoice.HOST specified from cross file, native file, or env var as ['/workspace/megous.com/apps-c/static-deps/output/arm64-lto/arm64-lto-pkg-config']
Trying pkg-config binary pkgconfig for machine MachineChoice.HOST at ['/workspace/megous.com/apps-c/static-deps/output/arm64-lto/arm64-lto-pkg-config']
WARNING: Found pkg-config '/workspace/megous.com/apps-c/static-deps/output/arm64-lto/arm64-lto-pkg-config' but it failed when run
```

Because pkg-config --something --version returns:

```
 ./arm64-lto-pkg-config --version
pkgconf: --version specified with other options or module names, assuming --modversion.
Please specify at least one package name on the command line.
```
I guess I can fix it by handling --version in the wrapper before calling pkg-config itself, but the check can be less strict anyway, as it was prior to 0.50.

**Comment by megous** [comment]
In particular it should be enough if pkg-config exists and executes, regardless of whether it returns status 0 or something else.

**Comment by dcbaker** [comment]
We do actually need to know what the version of pkg-config is because we use some arguments that were added to pkg-config relatively recently (at least recently enough that some LTS/EL Linux distros are too old), and in the future we might want to tailor the arguments we use based on the version. You like need to to add something like:
```sh
[[ "$1" == '--version' ]] && exec pkg-config --version
```
to the top of your shell script.

**Comment by megous** [comment]
Ok. That works. Thanks.

**Comment by kleisauke** [comment]
For reference, this was fixed in pkgconf v1.7.0 with commit https://github.com/pkgconf/pkgconf/commit/62bbd3b664d4b03011a4b382ed20353a91c30406.
